src/pybind/mgr/dashboard/frontend/src/app/shared/services/dep-checker.service.ts

```diff
@@ -18,19 +21,38 @@ export class DepCheckerService {
    * @param {string} actionDescription name of the action.
    * @param {string} itemDescription the item's name that the action operates on.
    * @param {Function} func the function to be called if orchestrator is available.
+   * @param {OrchestratorFeature[]} features Required orchestrator features for func to work.
    */
-  checkOrchestratorOrModal(actionDescription: string, itemDescription: string, func: Function) {
-    this.orchService.status().subscribe((status) => {
+  checkOrchestratorOrModal(
+    actionDescription: string,
+    itemDescription: string,
+    func: Function,
+    features?: OrchestratorFeature[]
+  ) {
+    this.orchService.status().subscribe((status: OrchestratorStatus) => {
+      let missingFeatures: OrchestratorFeature[];
       if (status.available) {
-        func();
-      } else {
-        this.modalService.show(OrchestratorDocModalComponent, {
-          initialState: {
-            actionDescription: actionDescription,
-            itemDescription: itemDescription
+        if (!_.isEmpty(features)) {
+          missingFeatures = _.filter(
+            features,
+            (feature) => !_.get(status.features, `${feature}.available`)
+          );
+          if (_.isEmpty(missingFeatures)) {
+            func();
+            return;
           }
-        });
+        } else {
+          func();
+          return;
+        }
       }
+      this.modalService.show(OrchestratorDocModalComponent, {
+        initialState: {
+          actionDescription: actionDescription,
+          itemDescription: itemDescription,
+          missingFeatures: missingFeatures
+        }
+      });
```

Comment:
Instead of showing a modal after the user tried to perform the action, wouldn't it be more user-friendly to simply disable those UI widgets (the reason for the element being disable could be explained from within a tooltip). As a user I really find frustrating when I'm allowed to do something and then immediately get a warning telling me that I cannot do that :/

Possible ideas:
* [HTTP verb OPTIONS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS) to get the supported methods for a given resource. That avoids the need of triggering 50x errors, which would pollute logs and may cause some unnecessary alarm when Network requests are checked.
* Keeping the cross-component approach by doing this in a directive (similar to what @nizamial09 is doing in https://github.com/ceph/ceph/pull/34545).

Reply:
> Instead of showing a modal after the user tried to perform the action, wouldn't it be more user-friendly to simply disable those UI widgets (the reason for the element being disable could be explained from within a tooltip). 

Thanks for the feedback!
The Create/Delete action of hosts are implemented like this before. I believe I got the suggestion to display a modal from the team when demonstrating OSD feature (create or delete) when orchestrator is not there. So all the related features are changed to displaying modal.

Reply:
Another advantage of displaying a modal is it allows us to embed more information. For example, a copy-able and click-able URL to the document.

Reply:
I agree with that approach for when the user can or must do something to enable a component (we are already doing it for RGW, Grafana, Prometheus), and I still think it's debatable.

In this case the user cannot do anything to enable those components, so the modal is basically leading them to perform 3 actions (click a useless button, then read a text, and finally click the close button)... just to get nothing. And even worse, we are not preventing users from (unintentionally) clicking again those UI elements.

It's the same as displaying the "Delete" button enabled when there's nothing to delete and opening a modal for informing users about the impossibility to delete things that don't exist and forcing them to click on the close button just to get where they were before.

What about a modeless approach, like [this Bootstrap's "Popovers for disabled elements"](https://getbootstrap.com/docs/4.0/components/popovers/#disabled-elements): it still allows interacting with the text from the popover (clicking and selecting text).

Reply:
Popovers look promising. After trying it for a while I found it needs more works and fine-tune, e.g.:
- It requires clicking. Hovering doesn't work.
- It seems not to work well with dropdown items. After clicking a dropdown item, the dorpdown is gone but the popover is there.

Can I create an issue for the popover suggestion? Refactoring the modal is a bit out of this PR's scope.

Reply:
Opss... Anyway my initial proposal was/is to simply disable those buttons and provide a tooltip with the reason.